src/main/java/ch/njol/skript/expressions/ExprEnchantmentOfferLevel.java

```diff
@@ -0,0 +1,113 @@
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.enchantments.EnchantmentOffer;
+import org.bukkit.event.Event;
+import org.bukkit.event.enchantment.PrepareItemEnchantEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Enchantment Offer Enchantment Level")
+@Description({"The enchantment level of an enchantment offer.",
+			"If the level is changed, it will always be at least 1.",
+			"This change is cosmetic, and does not change the number of levels applied.", 
+			"To change the number of levels applied, use the applied enchantment expression.",})
+@Examples("set enchantment level of enchantment offer 1 to 3")
+@Since("INSERT VERSION")
+@RequiredPlugins("1.11 or newer")
+public class ExprEnchantmentOfferLevel extends SimplePropertyExpression<EnchantmentOffer, Number>{
+
+	static {
+		if (Skript.classExists("org.bukkit.enchantments.EnchantmentOffer")) {
+			register(ExprEnchantmentOfferLevel.class, Number.class, "[enchant[ment]] level", "enchantmentoffers");
+		}
+	}
+
+	@Override
+	public Number convert(final EnchantmentOffer offer) {
+		return offer.getEnchantmentLevel();
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.REMOVE || mode == ChangeMode.REMOVE_ALL || mode == ChangeMode.RESET)
+			return null;
+		return CollectionUtils.array(Number.class);
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		EnchantmentOffer[] offers = getExpr().getArray(event);
+		if (offers.length == 0)
+			return;
+		int level = delta != null ? ((Number) delta[0]).intValue() : 1;
+		if (level < 1) level = 1;
+		int change = 1;
+		switch (mode) {
+			case SET:
+				for (EnchantmentOffer offer : offers)
+					offer.setEnchantmentLevel(level);
+				break;
+			case ADD:
+				for (EnchantmentOffer offer : offers) {
+					change = level + offer.getEnchantmentLevel();
+					if (change < 1)
+						change = 1;
+					offer.setEnchantmentLevel(change);
+				}
+				break;
+			case REMOVE:
+				for (EnchantmentOffer offer : offers) {
+					change = level - offer.getEnchantmentLevel();
+					if (change < 1)
+						change = 1;
+					offer.setEnchantmentLevel(change);
+				}
+				break;
+			case RESET:
+			case DELETE:
+			case REMOVE_ALL:
+				assert false;
+		}
+	}
+
+	@Override
+	public Class<Number> getReturnType() {
+		return Number.class;
+	}
+
+	@Override
+	protected String getPropertyName() {
+		return "[enchant[ment]] level";
```

Comment:
`getPropertyName` isn't meant to have the syntax tokens in it, it is used for the `toString` and should be representative of how it would read in a script